dss_metrics/minio_collector.py

```diff
@@ -0,0 +1,174 @@
+                    self.ustat_path + ' -p '
+                    + str(pid) + ' ' + str(self.seconds)
+                    + ' ' + str(self.num_iterations)
+                )
+                proc_file = os.path.join('/proc', str(pid), 'cmdline')
+                proc_cmd = 'minio'
+                with open(proc_file) as f:
+                    proc_cmd = f.readline()
+                minio_uuid = uuid.uuid3(uuid.NAMESPACE_DNS, proc_cmd)
+                proc = subprocess.Popen(cmd.split(' '), stdout=subprocess.PIPE)
+                stats_output['time'] = collection_time
+                minio_proc_map[minio_uuid] = proc
+            except Exception as error:
+                print(f'Caught exception while running USTAT {str(error)}')
+
+        device_subsystem_map = utils.get_device_subsystem_map()
+        for minio_uuid, proc in minio_proc_map.items():
+            while True:
+                line = proc.stdout.readline().decode('utf-8')
+                if not line:
+                    break
+                line = line.strip()
+                if line:
+                    try:
+                        full_key, value = line.split('=')
+                        whitelist_match = False
+
+                        # filter using whitelist patterns if required
+                        if self.filter:
+                            for regex in self.whitelist_patterns:
+                                if re.match(regex, full_key):
+                                    whitelist_match = True
+                                    break
+
+                        # check if value is a valid promotheus metric value
+                        valid_value_flag = value.replace('.', '').isdigit()
+
+                        fields = full_key.split('.')
+                        tags = {}
+                        tags['cluster_id'] = 'c01'
+                        tags['target_id'] = socket.gethostname()
+                        tags['minio_id'] = str(minio_uuid)
+                        tags['type'] = self.TYPE
+                        metric_name = fields[-1]
+
+                        if fields[0] == 'nkv':
+                            subsystem = fields[2]
+                            tags['subsystem_id'] = (
+                                device_subsystem_map[subsystem]['nqn']
+                            )
+                            if 'ip' in metric_name:
+                                valid_value_flag = False
+                        elif fields[0] == 'minio_upstream':
+                            tags['subsystem_id'] = 'minio_upstream'
+
+                        """
+                        XOR operation
+                        check if filter, then whitelist match should be True
+                        if not filter, than whitelist match should be False
+                        """
+                        if (
+                            valid_value_flag
+                            and (self.filter == whitelist_match)
```

Comment:
brackets are not needed
Also don't complicate code by breaking the condition logic into multiple lines unless it is a standard.
Try to make the whole if loop into  if (cond):